evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -235,7 +200,6 @@ def predict(self, X, objective=None, X_train=None, y_train=None):
         X.index = self._move_index_forward(
             X_train.index[-X.shape[0] :], self.gap + X.shape[0]
         )
-        self._validate_holdout_datasets(X, X_train)
```

Comment:
Based on the third point of [this](https://github.com/alteryx/evalml/issues/3078#issuecomment-1010307947): Rather than raising a ValueError in predict, let's refactor that logic into a helper function that could be used before calling predict. predict should no longer raise exceptions if the data violates our constraints.